app/v2/controllers/V2MovementsController.scala

```diff
@@ -398,13 +390,24 @@ class V2MovementsControllerImpl @Inject() (
       implicit request =>
         implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
 
-        persistenceService
-          .getMessages(request.eoriNumber, movementType, movementId, receivedSince, page, count, receivedUntil)
-          .asPresentation
-          .fold(
-            presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
-            response => Ok(Json.toJson(HateoasMovementMessageIdsResponse(movementId, response, receivedSince, movementType, page, count, receivedUntil)))
-          )
+        val pageValidation: EitherT[Future, PresentationError, Unit] =
+          ensurePositive(page.map(_.value), "page")
+
+        val countValidation: EitherT[Future, PresentationError, Unit] =
+          ensurePositive(count.map(_.value), "count")
```

Comment:
Don't make these variables, use the directly in the for comprehension. If you do that, if the first fails, the second won't execute. No point wasting cycles.

Reply:
I was thinking the same. Also I think if he is going to change then **result Val** also can be removed and can be Folded just like earlier it was done.